app/src/main/java/com/imashnake/animite/features/media/MediaPage.kt

```diff
@@ -90,124 +91,120 @@ fun MediaPage(
             modifier = Modifier.background(MaterialTheme.colorScheme.background)
         ) {
             Box(
                 Modifier
                     .fillMaxSize()
                     .verticalScroll(scrollState)
             ) {
-                MediaBanner(
-                    imageUrl = media.bannerImage,
-                    tintColor = Color(media.color ?: 0).copy(alpha = 0.25f),
-                    modifier = Modifier
-                        .height(bannerHeight)
-                        .fillMaxWidth()
-                        .bannerParallax(scrollState)
-                )
-
-                Column(
-                    modifier = Modifier
-                        .fillMaxHeight()
-                        .padding(top = bannerHeight)
-                        .background(MaterialTheme.colorScheme.background)
-                        .padding(bottom = LocalPaddings.current.large)
-                        .navigationBarsPadding(),
-                    verticalArrangement = Arrangement.spacedBy(LocalPaddings.current.large)
-                ) {
-                    MediaDetails(
-                        title = media.title.orEmpty(),
-                        description = media.description.orEmpty(),
-                        // TODO: Can we do something about this Modifier chain?
-                        //  Fix this in a follow up PR with `BannerLayout`.
-                        modifier = Modifier
-                            .padding(
-                                start = LocalPaddings.current.large
-                                        + dimensionResource(R.dimen.media_card_width)
-                                        + LocalPaddings.current.large,
-                                top = LocalPaddings.current.medium,
-                                end = LocalPaddings.current.large
-                            )
-                            .landscapeCutoutPadding()
-                            .height(
-                                WindowInsets.statusBars
-                                    .asPaddingValues()
-                                    .calculateTopPadding()
-                                        + dimensionResource(R.dimen.media_card_top_padding)
-                                        + dimensionResource(R.dimen.media_card_height)
-                                        - dimensionResource(R.dimen.banner_height)
-                                        - LocalPaddings.current.medium
-                            )
-                            .fillMaxSize()
-                    )
-
-                    if (!media.ranks.isNullOrEmpty()) {
-                        MediaRankings(
-                            rankings = media.ranks,
+                BannerLayout(
+                    banner = { bannerModifier ->
+                        MediaBanner(
+                            imageUrl = media.bannerImage,
+                            tintColor = Color(media.color ?: 0).copy(alpha = 0.25f),
```

Comment:
Can we have `media` fall back to black or something before it makes its way to the UI? Non-nullable color would be very convenient

Reply:
https://github.com/imashnake0/Animite/blob/6e3eda11885b23bed2a81f2028647f57e2762130/app/src/main/java/com/imashnake/animite/features/media/MediaPageViewModel.kt#L33 will be fixed in a different PR